elastic/integrations
# Add synthetics package

## What does this PR do?

The goal of this PR is to add a new package for Elastic synthetics. The eventual target is to be able to use this package in conjunction with Elastic Agent to index Heartbeat documents for a simple HTTP check.

The acceptance criteria of this is an MVP HTTP uptime monitor experience. This includes defining a monitor name, URL, and period. Heartbeat should be able to accept the data stream info and utilize the resources provided by the package to achieve this.

## Checklist


## Author's Checklist


- [ ] this package is
- [ ] this package conforms to spec requirements
- [ ] this package creates integrations that agent can consume
- [ ] this package provides the necessary resources for agent to index heartbeat documents

## How to test this PR locally

- configure a basic HTTP check integration using the package
- enroll an agent, verify it doesn't encounter errors regarding the integration
- verify the required index pattern/mapping is created
- open the Uptime app, it should handle the documents Agent is indexing
